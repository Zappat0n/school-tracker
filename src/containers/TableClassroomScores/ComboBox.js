import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';
import { postEvent } from '../../api/queries';

const ComboBox = ({ _id, presentation, student }) => {
  const token = useSelector((state) => state.user.token);
  // const dispatch = useDispatch();
  let id = _id;

  const getScore = (value) => {
    switch (value) {
      case ' ': return 0;
      case '/': return 1;
      case 'ꓥ': return 2;
      case '𐊅': return 3;
      default: return null;
    }
  };

  async function query(score) {
    if (!token) return;
    const response = await postEvent(token, new Date().toISOString().split('T')[0], student, presentation, getScore(score));
    console.log(response);
    if (response && response.data) id = response;
  }

  const handleChange = (e) => {
    e.preventDefault();
    query(e.target.value);
  };

  return (
    <td>
      <select className="combo" id={id} presentation={presentation} student={student} onChange={(e) => handleChange(e)}>
        <option value=" "> </option>
        <option value="/">/</option>
        <option value="ꓥ">ꓥ</option>
        <option value="𐊅">𐊅</option>
      </select>
    </td>
  );
};

ComboBox.defaultProps = {
  _id: undefined,
};

ComboBox.propTypes = {
  presentation: PropTypes.string.isRequired,
  student: PropTypes.instanceOf(Array).isRequired,
  _id: PropTypes.number,
};

export default ComboBox;
