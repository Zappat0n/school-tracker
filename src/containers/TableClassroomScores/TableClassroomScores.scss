@import "../../index.scss";

.container-table-scores {
  @include responsive-scores;
  margin: 0 auto;
}

.table {
  @media (max-width: 768px) {
    overflow:scroll;
  }
}

.table-classroom-head {
  th {
    @include small-font;
  }
  .head-link {
    color: white;
    font-weight: 700;
  }
}

.table-classroom-body {

  td, .combo {
    @include small-font;

  }

  tr:nth-child(even) {
    background-color: #f8f6ff;
  }

  .row:hover {
    background: lightgray !important;
  }
}
