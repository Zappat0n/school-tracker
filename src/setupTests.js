// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import { createStore, combineReducers } from 'redux';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import userReducer from './slices/userSlice';
import classroomTableReducer from './slices/classroomTableSlice';

const rootReducer = combineReducers({
  user: userReducer, classroomTable: classroomTableReducer,
});

const initialState = {
  user: {
    tables: {},
    title: '',
  },
  classroomTable: {
    students: [],
    presentations: [],
    scores: {},
  },
};

const store = createStore(rootReducer, initialState);

const renderComponent = (component) => render(
  <Provider store={store}>
    <BrowserRouter>
      {component}
    </BrowserRouter>
  </Provider>,
);

const rendererForComponent = (component) => renderer.create(
  <Provider store={store}>
    <BrowserRouter>
      {component}
    </BrowserRouter>
  </Provider>,
);

export { renderComponent, rendererForComponent, store };
