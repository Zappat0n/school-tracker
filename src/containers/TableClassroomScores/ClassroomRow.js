import PropTypes from 'prop-types';
import ComboBox from './ComboBox';

const ClassroomRow = ({ presentation, students, handleChange }) => (
  <>
    <tr className="row">
      <td>{presentation.name}</td>
      {
      (students || []).map((student) => (
        <ComboBox
          key={student.id}
          presentation={presentation.id}
          student={student.id}
          handleChange={(event, presentation, student, id) => {
            handleChange(event, presentation, student, id);
          }}
        />
      ))
      }
    </tr>
  </>
);

ClassroomRow.propTypes = {
  presentation: PropTypes.instanceOf(Object).isRequired,
  students: PropTypes.instanceOf(Object).isRequired,
  handleChange: PropTypes.func.isRequired,
};

export default ClassroomRow;
