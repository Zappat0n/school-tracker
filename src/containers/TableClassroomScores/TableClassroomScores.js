import PropTypes from 'prop-types';
import { useDispatch, useSelector } from 'react-redux';
import { addTable, changeTitle } from '../../slices/userSlice';
import { getIndex } from '../../api/queries';
import ClassroomRow from './ClassroomRow';
import './TableClassroomScores.scss';

const TableClassroomScores = ({ id, title }) => {
  const request = `classrooms/${id}/scores/`;
  const table = useSelector((state) => state.user.tables[request]);
  const token = useSelector((state) => state.user.token);
  const dispatch = useDispatch();

  dispatch(changeTitle(title));

  async function query() {
    if (!token) return;
    const response = {
      tableName: request,
      data: await getIndex(token, request),
    };
    if (response && response.data) dispatch(addTable(response));
  }

  if (!table) query();
  console.log(table);
  return (
    <div className="container-table">
      {(table && table.data) ? (
        <div className="table">
          <table>
            <thead className="table-classroom-head">
              <tr>
                {(table.data.students ? ['Presentations'].concat(table.data.students) : []).map((value, index) => (
                  <th className={`column${index + 1}`} key={value.name}>{value.name}</th>
                ))}
              </tr>
            </thead>
            <tbody className="table-body">
              {
              (table.data.presentations ? table.data.presentations.slice(1, 10) : []).map(
                (value) => (
                  <ClassroomRow
                    key={value.name}
                    id={value.id}
                    presentation={value.name}
                    students={table.data.students.map((value) => value.id)}
                  />
                ),
              )
              }
              {/* table.data.map((row) => <Row key={row.id} data={row} />) */}
            </tbody>
          </table>
        </div>
      ) : ''}
    </div>
  );
};

TableClassroomScores.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

export default TableClassroomScores;
