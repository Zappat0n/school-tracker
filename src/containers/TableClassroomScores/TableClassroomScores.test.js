import React from 'react';
import '@testing-library/jest-dom';
import { screen } from '@testing-library/react';
import { renderComponent, rendererForComponent, store } from '../../setupTests';
import TableClassroomScores from './TableClassroomScores';
import {
  saveScores, saveStudents, savePresentations,
} from '../../slices/classroomTableSlice';

test('Display title', () => {
  store.dispatch(saveStudents([
    { id: 1, name: 'John' },
    { id: 2, name: 'Peter' },
    { id: 3, name: 'Mark' },
    { id: 4, name: 'Louis' },
  ]));

  store.dispatch(savePresentations([
    { id: 1, name: 'Math' },
    { id: 2, name: 'Language' },
    { id: 3, name: 'Science' },
    { id: 4, name: 'Spanish' },
  ]));

  store.dispatch(saveScores([
    {
      id: 1, presentation_id: 1, student_id: 1, score: 1,
    },
    {
      id: 2, presentation_id: 2, student_id: 2, score: 2,
    },
    {
      id: 3, presentation_id: 1, student_id: 3, score: 3,
    },
    {
      id: 4, presentation_id: 2, student_id: 2, score: 0,
    },
  ]));

  renderComponent(
    <>
      <TableClassroomScores tableName="classrooms" id="1" title="Classrooms" />
    </>,
  );

  expect(screen.getByText('Math')).toBeInTheDocument();
  expect(screen.getByText('Language')).toBeInTheDocument();
  expect(screen.getByText('Science')).toBeInTheDocument();
  expect(screen.getByText('Spanish')).toBeInTheDocument();
  expect(screen.getByText('John')).toBeInTheDocument();
  expect(screen.getByText('Peter')).toBeInTheDocument();
  expect(screen.getByText('Mark')).toBeInTheDocument();
  expect(screen.getByText('Louis')).toBeInTheDocument();
});

test('Stock Match SnapShot', () => {
  const tree = rendererForComponent(
    <>
      <TableClassroomScores tableName="classrooms" id="1" title="Classrooms" />
    </>,
  );
  expect(tree).toMatchSnapshot();
});
