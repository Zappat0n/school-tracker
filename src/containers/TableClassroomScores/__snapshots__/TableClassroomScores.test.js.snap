// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stock Match SnapShot 1`] = `
<div
  className="container-table-scores"
>
  <div
    className="table"
  >
    <table>
      <thead
        className="table-classroom-head"
      >
        <tr>
          <th
            className="column1"
          />
          <th
            className="column2"
          >
            John
          </th>
          <th
            className="column3"
          >
            Peter
          </th>
          <th
            className="column4"
          >
            Mark
          </th>
          <th
            className="column5"
          >
            Louis
          </th>
        </tr>
      </thead>
      <tbody
        className="table-body"
      >
        <tr
          className="row"
        >
          <td>
            Math
          </td>
          <td>
            <select
              className="combo"
              id={1}
              onChange={[Function]}
              presentation={1}
              student={1}
              value="/"
            >
              <option
                value=" "
              >
                 
              </option>
              <option
                value="/"
              >
                /
              </option>
              <option
                value="Íì•"
              >
                Íì•
              </option>
              <option
                value="êäÖ"
              >
                êäÖ
              </option>
            </select>
          </td>
          <td>
            <select
              className="combo"
              onChange={[Function]}
              presentation={1}
              student={2}
              value=""
            >
              <option
                value=" "
              >
                 
              </option>
              <option
                value="/"
              >
                /
              </option>
              <option
                value="Íì•"
              >
                Íì•
              </option>
              <option
                value="êäÖ"
              >
                êäÖ
              </option>
            </select>
          </td>
          <td>
            <select
              className="combo"
              id={3}
              onChange={[Function]}
              presentation={1}
              student={3}
              value="êäÖ"
            >
              <option
                value=" "
              >
                 
              </option>
              <option
                value="/"
              >
                /
              </option>
              <option
                value="Íì•"
              >
                Íì•
              </option>
              <option
                value="êäÖ"
              >
                êäÖ
              </option>
            </select>
          </td>
          <td>
            <select
              className="combo"
              onChange={[Function]}
              presentation={1}
              student={4}
              value=""
            >
              <option
                value=" "
              >
                 
              </option>
              <option
                value="/"
              >
                /
              </option>
              <option
                value="Íì•"
              >
                Íì•
              </option>
              <option
                value="êäÖ"
              >
                êäÖ
              </option>
            </select>
          </td>
        </tr>
        <tr
          className="row"
        >
          <td>
            Language
          </td>
          <td>
            <select
              className="combo"
              onChange={[Function]}
              presentation={2}
              student={1}
              value=""
            >
              <option
                value=" "
              >
                 
              </option>
              <option
                value="/"
              >
                /
              </option>
              <option
                value="Íì•"
              >
                Íì•
              </option>
              <option
                value="êäÖ"
              >
                êäÖ
              </option>
            </select>
          </td>
          <td>
            <select
              className="combo"
              id={4}
              onChange={[Function]}
              presentation={2}
              student={2}
              value=" "
            >
              <option
                value=" "
              >
                 
              </option>
              <option
                value="/"
              >
                /
              </option>
              <option
                value="Íì•"
              >
                Íì•
              </option>
              <option
                value="êäÖ"
              >
                êäÖ
              </option>
            </select>
          </td>
          <td>
            <select
              className="combo"
              onChange={[Function]}
              presentation={2}
              student={3}
              value=""
            >
              <option
                value=" "
              >
                 
              </option>
              <option
                value="/"
              >
                /
              </option>
              <option
                value="Íì•"
              >
                Íì•
              </option>
              <option
                value="êäÖ"
              >
                êäÖ
              </option>
            </select>
          </td>
          <td>
            <select
              className="combo"
              onChange={[Function]}
              presentation={2}
              student={4}
              value=""
            >
              <option
                value=" "
              >
                 
              </option>
              <option
                value="/"
              >
                /
              </option>
              <option
                value="Íì•"
              >
                Íì•
              </option>
              <option
                value="êäÖ"
              >
                êäÖ
              </option>
            </select>
          </td>
        </tr>
        <tr
          className="row"
        >
          <td>
            Science
          </td>
          <td>
            <select
              className="combo"
              onChange={[Function]}
              presentation={3}
              student={1}
              value=""
            >
              <option
                value=" "
              >
                 
              </option>
              <option
                value="/"
              >
                /
              </option>
              <option
                value="Íì•"
              >
                Íì•
              </option>
              <option
                value="êäÖ"
              >
                êäÖ
              </option>
            </select>
          </td>
          <td>
            <select
              className="combo"
              onChange={[Function]}
              presentation={3}
              student={2}
              value=""
            >
              <option
                value=" "
              >
                 
              </option>
              <option
                value="/"
              >
                /
              </option>
              <option
                value="Íì•"
              >
                Íì•
              </option>
              <option
                value="êäÖ"
              >
                êäÖ
              </option>
            </select>
          </td>
          <td>
            <select
              className="combo"
              onChange={[Function]}
              presentation={3}
              student={3}
              value=""
            >
              <option
                value=" "
              >
                 
              </option>
              <option
                value="/"
              >
                /
              </option>
              <option
                value="Íì•"
              >
                Íì•
              </option>
              <option
                value="êäÖ"
              >
                êäÖ
              </option>
            </select>
          </td>
          <td>
            <select
              className="combo"
              onChange={[Function]}
              presentation={3}
              student={4}
              value=""
            >
              <option
                value=" "
              >
                 
              </option>
              <option
                value="/"
              >
                /
              </option>
              <option
                value="Íì•"
              >
                Íì•
              </option>
              <option
                value="êäÖ"
              >
                êäÖ
              </option>
            </select>
          </td>
        </tr>
        <tr
          className="row"
        >
          <td>
            Spanish
          </td>
          <td>
            <select
              className="combo"
              onChange={[Function]}
              presentation={4}
              student={1}
              value=""
            >
              <option
                value=" "
              >
                 
              </option>
              <option
                value="/"
              >
                /
              </option>
              <option
                value="Íì•"
              >
                Íì•
              </option>
              <option
                value="êäÖ"
              >
                êäÖ
              </option>
            </select>
          </td>
          <td>
            <select
              className="combo"
              onChange={[Function]}
              presentation={4}
              student={2}
              value=""
            >
              <option
                value=" "
              >
                 
              </option>
              <option
                value="/"
              >
                /
              </option>
              <option
                value="Íì•"
              >
                Íì•
              </option>
              <option
                value="êäÖ"
              >
                êäÖ
              </option>
            </select>
          </td>
          <td>
            <select
              className="combo"
              onChange={[Function]}
              presentation={4}
              student={3}
              value=""
            >
              <option
                value=" "
              >
                 
              </option>
              <option
                value="/"
              >
                /
              </option>
              <option
                value="Íì•"
              >
                Íì•
              </option>
              <option
                value="êäÖ"
              >
                êäÖ
              </option>
            </select>
          </td>
          <td>
            <select
              className="combo"
              onChange={[Function]}
              presentation={4}
              student={4}
              value=""
            >
              <option
                value=" "
              >
                 
              </option>
              <option
                value="/"
              >
                /
              </option>
              <option
                value="Íì•"
              >
                Íì•
              </option>
              <option
                value="êäÖ"
              >
                êäÖ
              </option>
            </select>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
